### ISSUE TITLE
New comment on GitLab

### ISSUE THREAD

**Issue by AlexDenisov** [danger/danger#1084]
# Report

## What did you do?

I ask Danger to always create new comments:

```
danger --new-comment --fail-on-errors=true --dangerfile=ci/danger/lint.rb
```

## What did you expect to happen?

For each re-run of our CI expected to see a new comment, preserving the old one.
I.e., the merge request history should look like this:

```
commit: abc
danger: please fix typos
commit: def
danger: looks good to me
```

## What happened instead?

The first comment' body is replaced with a new comment.
I.e., the merge request history looks like this:

```
commit: abc
danger: please fix typos
```

And after new commit:

```
commit: abc
danger: looks good to me
commit: def
```

## Your Environment

* Which CI are you running on?

  Jenkins

* Are you running the latest version of Danger?

  5.14.0

* What is your Dangerfile?

  ```ruby
  require 'open3'

  def run(title, command)
    stdout, stderr, status = Open3.capture3(command)
    if status.success?
      message "#{title} passed"
    else
      failure #{title} failed
    end
  end

  run('sanity checks', 'make sanity_checks')
  ```

## Additional info

The fix for my expected behaviour is trivial (see below), but the question is whether it is a bug or I misunderstand the purpose of the `--new-comment` flag?

```diff
diff --git a/lib/danger/request_sources/gitlab.rb b/lib/danger/request_sources/gitlab.rb
index 644bb9e1..014a8022 100644
--- a/lib/danger/request_sources/gitlab.rb
+++ b/lib/danger/request_sources/gitlab.rb
@@ -133,7 +133,7 @@ def update_pull_request!(warnings: [], errors: [], messages: [], markdowns: [],
                                  danger_id: danger_id,
                                   template: "gitlab")
 
-          if editable_comments.empty?
+          if editable_comments.empty? or should_create_new_comment
             client.create_merge_request_comment(
               ci_source.repo_slug, ci_source.pull_request_id, body
             )
```

Thank you for the great tool, it helps a lot!

**Comment by orta** [comment]
Yep, I agree, I don't think it was probably added right to the GitLab side looking at it's usage in comparison to GitHub

**Comment by AlexDenisov** [comment]
Shall I submit the patch then?

**Comment by orta** [comment]
Yep, sure 👍

**Comment by AlexDenisov** [comment]
Resolved via #1085
